### PR TITLE
Add snaps

### LINKED ISSUE #1776: Snaps for Gaphor
Ubuntu comes with snaps and most of the gnome things comes with snaps so I think will be great to have it. Canonical released a pdf about using Gnome extensions https://blogs.gnome.org/hellsworth/files/2019/11/GNOME-extensions-for-desktop-apps.pdf

I'm happy to help if needed.

### PR DESCRIPTION
**Pull request by Saphyel**
close #1776 

Unfortunately due to personal reason I couldn't work on this as much as I wanted but this is half working :(

### CODE REVIEW

**Comment by amolenaar** [comment]
This is pretty neat :+1:. What are the steps to run it and build a snap for testing? Do you have any idea (or reference) on how to make the snap available to the wider audience?

**Comment by Saphyel** [comment]
To build the package you need to execute:
```
snapcraft
```

This will generate a snap file (with a name like: `gaphor_0.1_amd64.snap` in my case but there's an architecture section to add more) in the root directory, so if you want to install it locally you need to run:
```
sudo snap install file
```

For more information about how to debug/test (the step missing at the moment to continue) https://snapcraft.io/docs/debug-snaps

For publishing it will require some credentials for logging and then it is only upload https://snapcraft.io/docs/releasing-your-app

Some of my notes:
* Building from the source (.) is way faster than using the git (probably because of the size of the repository? Not entirely sure if this a problem of snaps or something I'm doing wrong)
* I can't make work the `apps` and `slots` section, probably I'm missing something. I requested them to update the docs for using core22 and python
* Docker and snapcraft are enemies

**Comment by amolenaar** [comment]
Is it possible to use GNOME 43 as a base in snap?
We're using libadwaita 1.2 features. Also GTK 4.8, which ships with GNOME 43, has a bunch of fixes in the tree view, which are sort of required for Gaphor.

**Comment by Saphyel** [comment]
@amolenaar I'll look into it. I know it can use GTK4 (Gnome 42) but  43 I'm not sure, the documentation about it is here => https://snapcraft.io/docs/gnome-extension

If it is not possible or they are not planning to allow using newer version soon I guess the only option is to wait until core24 :(

**Comment by danyeaw** [comment]
If Snaps aren't going to support new Gnome releases immediately, this sounds like sort of a deal killer to me. Unfortunately, we will have to spend more effort keeping backwards compatibility with another packaging format.

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: gaphor
+base: core22
+version: '0.1'
+summary: Simple modeling tool
+description: |
+  Gaphor is a simple modeling tool written in Python. It is Free and Open Source software
```
Comment (vanillajonathan) [inline snap/snapcraft.yaml:6]:
```suggestion
  Gaphor is a simple modeling tool written in Python. It is free and open source software
```

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    plugs: [home, network]
+
+parts:
+  gaphor:
+    plugin: python
+    python-packages: [poetry]
+    # source: .
+    source: git@github.com:gaphor/gaphor.git
+    source-type: git
+    source-tag: '2.9.2'
```
Comment (vanillajonathan) [inline snap/snapcraft.yaml:31]:
```suggestion
    source-tag: '2.17.0'
```

**Comment by vanillajonathan** [comment]
Add `license: Apache-2.0`.

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+name: gaphor
+base: core22
+version: '0.1'
+summary: Simple modeling tool
+description: |
+  Gaphor is a simple modeling tool written in Python. It is Free and Open Source software
+  for creating software and system architecture and designs. It is aimed at beginning modelers.
+grade: devel # must be 'stable' to release into candidate/stable channels
```
Comment (vanillajonathan) [inline snap/snapcraft.yaml:8]: Add:
```
issues: https://github.com/gaphor/gaphol/issues
source-code: https://github.com/gaphor/gaphor
contact: https://gaphor.org/discuss/
website: https://gaphor.org/
license: Apache-2.0
```

File: snap/snapcraft.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+confinement: strict
```
Comment (vanillajonathan) [inline snap/snapcraft.yaml:9]:
```suggestion
confinement: strict
icon: logos/org.gaphor.Gaphor.svg
```

**Comment by Saphyel** [comment]
Sorry I can't spend time on snap at the moment (my ubuntu machine is broken).

@vanillajonathan  feel free to copy and start a new PR or continue this one

File: snap/snapcraft.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+name: gaphor
+base: core22
+version: '0.1'
+summary: Simple modeling tool
+description: |
+  Gaphor is a simple modeling tool written in Python. It is free and open source software
+  for creating software and system architecture and designs. It is aimed at beginning modelers.
+grade: devel # must be 'stable' to release into candidate/stable channels
+issues: https://github.com/gaphor/gaphol/issues
```
Comment (vanillajonathan) [inline snap/snapcraft.yaml:9]:
```suggestion
issues: https://github.com/gaphor/gaphol/issues
issues: https://github.com/gaphor/gaphor/issues
```

**Comment by vanillajonathan** [comment]
> Sorry I can't spend time on snap at the moment (my ubuntu machine is broken).

You can install Ubuntu on Windows using WSL.
https://apps.microsoft.com/store/detail/ubuntu-on-windows/9NBLGGH4MSV6

**Comment by danyeaw** [comment]
I left a comment in #1776, we are going to focus on Linux packaging on Flatpak. :+1:

**Pull request closed without merging**